revolutionpi/digital_interrupts.go

```diff
@@ -75,16 +80,22 @@ func initializeDigitalInterrupt(pin SPIVariable, g *gpioChip) (*digitalInterrupt
 
 	// check if the pin is configured as a counter
 	// b[0] == 0 means the interrupt is disabled, b[0] == 3 means the pin is configured for encoder mode
-	if b[0] == 0 || b[0] == 3 {
-		return &digitalInterrupt{}, errors.New("pin is not configured as a counter")
+	if b[0] == 0 || (b[0] == 3 && !isEncoder) {
+		return &digitalInterrupt{}, fmt.Errorf("pin %s is not configured as a counter", di.pinName)
 	}
+
 	di.enabled = true
+	di.isEncoder = b[0] == 3
 
 	return &di, nil
 }
 
+func (di *diWrapper) Value(ctx context.Context, extra map[string]interface{}) (int64, error) {
+	return di.pin.Value()
+}
+
 // Note: The revolution pi only supports uint32 counters, while the Value API expects int64.
-func (di *digitalInterrupt) Value(ctx context.Context, extra map[string]interface{}) (int64, error) {
+func (di *digitalInterrupt) Value() (int64, error) {
```

Comment:
This no longer matches the GRPC interface to get an interrupt's value. I realize there's a layer of indirection between the protobufs and this function where we can throw away the extra data, but it feels like a brittle design (if you want to start doing fancier things in the future, you need to put this line back the way it used to be). What's the reasoning behind changing this?

Reply:
pretty much what you said tbh. We weren't using context and extra so I felt like we could simplify the signature. I can see a world where we might want the context, but since `extra` is pretty much only meant to hack in extra methods I did not believe it was necessary to keep it.

Reply:
:shrug: I can see a future where you need to undo this change, and can also see a future where this is fine as-is. I trust your judgment; make it how you want.